internal/pkg/agent/application/info/inject_config_windows.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package info
+
+import (
+	"fmt"
+	"runtime"
+	"syscall"
+	"unsafe"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/go-sysinfo"
+)
+
+// InjectAgentConfig injects config to a provided configuration.
+func InjectAgentConfig(c *config.Config) error {
+	globalConfig, err := agentGlobalConfig()
+	if err != nil {
+		return err
+	}
+
+	if err := c.Merge(globalConfig); err != nil {
+		return errors.New("failed to inject agent global config", err, errors.TypeConfig)
+	}
+
+	return nil
+}
+
+// agentGlobalConfig gets global config used for resolution of variables inside configuration
+// such as ${path.data}.
+func agentGlobalConfig() (map[string]interface{}, error) {
+	hostInfo, err := sysinfo.Host()
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
+		"path": map[string]interface{}{
+			"data":   paths.Data(),
+			"config": paths.Config(),
+			"home":   paths.Home(),
+			"logs":   paths.Logs(),
+		},
+		"host": map[string]interface{}{
+			"id": hostInfo.Info().UniqueID,
+		},
+		"runtime.os":             runtime.GOOS,
+		"runtime.arch":           runtime.GOARCH,
+		"runtime.native_arch":    nativeArch(),
```

Comment:
Rather than duplicating the entire map here, you should be able to call the non-windows (or lets say the common) version and then just add the single additional `runtime.native_arch` key.

For example:

```go
globalCfg, err := agentGlobalConfig()
if err != nil {
    return fmt.Errorf("getting common global config: %w", err)
}

globalCfg["runtime.native_arch"] = nativeArch()

return globalCfg
```

Essentially just augmenting the common configuration with the windows key, rather than duplicating it for both cases.

Reply:
thanks, indeed. I'm also now wondering if this should rather be added to the go-sysinfo package

Reply:
I can't think of a reason why you couldn't put it in go-sysinfo, that will allow us to reuse this outside the agent.

Easiest thing to do is open a PR there.

Reply:
I've found out that go-sysinfo is only extending golang.org/x/sys/windows, where the latter already includes `IsWow64Process2` so I use it directly instead of loading the dll.